roles/db-copy/tasks/active-copy.yml

```diff
@@ -0,0 +1,137 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+- name: Active-copy | Check pmon process
+  shell: "set -o pipefail; ps -ef | ( grep pmon || true ) | ( grep -i {{ db_name }} || true ) | ( grep -v grep || true ) | wc -l"
+  changed_when: false
+  register: pmon_proc
+  tags: active-duplicate
+
+- name: Active-copy | Add oratab entry
+  lineinfile:
+    path: /etc/oratab
+    regexp: '^{{ oracle_sid }}\:'
+    line: '{{ oracle_sid }}:{{ oracle_home }}:N'
+    owner: "{{ oracle_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Set random password
+  set_fact:
+    sys_pass: "{{ lookup('password', '/dev/null length=16 chars=ascii_letters,digits') }}#_"
+  tags: active-duplicate
+
+- name: Active-copy | Set sys password for primary db
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/orapwd file={{ oracle_home }}/dbs/orapw{{ db_name }} force=y password={{ sys_pass }}
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  delegate_to: primary1
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Add static listener entry
+  lineinfile:
+    path: "{{ grid_home }}/network/admin/listener.ora"
+    regexp: "^SID_LIST_{{ listener_name }}"
+    line: "SID_LIST_{{ listener_name }}=(SID_LIST=(SID_DESC=(GLOBAL_DBNAME={{ oracle_sid }}{{ standby_suffix }}.{{ db_domain }})(ORACLE_HOME={{ oracle_home }})(SID_NAME={{ oracle_sid }})))"
```

Comment:
What if db_domain is empty?  Is it OK to have a trailing dot?

Reply:
good catch! will change it to omit if no db_domain

Reply:
Maybe create a new variable for the fully-qualified DB name and reference it as needed?

Reply:
I replaced variables with if conditions for now. For the full qualified db name it worth to have a variable however the change would be related to many different files and I would rather create a new change to cover it